app/internal/import_file.py

```diff
@@ -41,185 +45,420 @@
-            if not is_valid_data_event_ics(component):
-                return []
-            save_calendar_content_ics(component, calendar_content)
+def _get_data_from_txt_file(txt_file_path: str) -> List[Dict[str, Any]]:
+    """Returns a list of event data in dictionaries from a *.txt file.
+
+    Args:
+        txt_file_path: The file path.
+
+    Returns:
+        A list of event data in dictionaries, or an empty list
+        if the data is not valid.
+
+    """
+    calendar_content: List[Dict[str, Any]] = []
+    for event in _get_event_from_txt_file(txt_file_path):
+        if not _is_event_text_valid(event):
+            return []
+
+        event_data = _get_event_data_from_text(event)
+
+        if not _is_event_dates_valid(
+                event_data["start_date"], event_data["end_date"]):
+            return []
+
+        _add_event_component_txt(event_data, calendar_content)
+
     return calendar_content
 
 
-def save_events_to_database(events: List[Dict[str, Union[str, Any]]],
-                            user_id: int,
-                            session: Session) -> None:
-    """insert the events into Event table"""
+def _get_event_from_txt_file(txt_file: str) -> Iterator[str]:
+    """Opens a *.txt file and returns a row of event data from it.
+
+    Args:
+        txt_file: The file path.
+
+    Yields:
+        A row of event data.
+
+    """
+    with open(txt_file, "r") as text:
+        for row in text.readlines():
+            yield row
+
+
+def _is_event_text_valid(text: str) -> bool:
+    """Whether the event text contains valid data.
+
+    Args:
+        text: The event text.
+
+    Returns:
+        True if valid, otherwise returns False.
+
+    """
+    get_values = EVENT_PATTERN.search(text)
+    get_values2 = EVENT_PATTERN2.search(text)
+    return get_values is not None or get_values2 is not None
+
+
+def _get_event_data_from_text(text: str) -> Dict[str, Any]:
+    """Returns the event data from the text.
+
+    Args:
+        text: The event text.
+
+    Returns:
+        A dictionary with the event data.
+
+    """
+    if len(text.split(", ")) == 5:
+        head, content, start_date, end_date, location = text.split(", ")
+        location = location.replace("\n", "")
+    else:
+        head, content, start_date, end_date = text.split(", ")
+        end_date = end_date.replace("\n", "")
+        location = ""
+
+    return {
+        "head": head,
+        "content": content,
+        "start_date": start_date,
+        "end_date": end_date,
+        "location": location,
+    }
+
+
+def _is_event_dates_valid(start: str, end: str) -> bool:
+    """Whether the event dates are valid.
+
+    Args:
+        start: The start time of the event.
+        end: The end time of the event.
+
+    Returns:
+        True if valid, otherwise returns False.
+
+    """
+    start_date = _convert_string_to_date(start)
+    end_date = _convert_string_to_date(end)
+    if not start_date and not end_date:
```

Comment:
Prefer `if start_date is None or end_date is None:`

Reply:
Done.